Document how to run dev env

dpordomingo opened the pull request:
Document how to run dev

The current way to run development environment with hot reloading uses the `srcd/sourced-ui` image required by `sourced` active config, which may be incorrect many times (if that version, and the current sources of `srcd-/sourced-ui` being tested mismatch in an incompatible way)

The proper way to run the dev environment requires to build the `srcd/sourced-ui:latest-dev` development image from sources and then use it for the hot reloading environment

---

<!-- Please leave this template at the end of your description, checking the option that applies -->

* [x] I have updated the CHANGELOG file according to the conventions in [keepachangelog.com](https://keepachangelog.com)
* [ ] This PR contains changes that do not require a mention in the CHANGELOG file

carlosms commented on CONTRIBUTING.md:
```diff
@@ -137,6 +145,11 @@ if needed)
     The first time you launch it, it will take some time to build all the UI assets,
     you can see the progress of the build from `sourced-ui` logs (see next step).
 
+    It may happen that the hot reloading stop working with an error message
```
```suggestion
    It may happen that the hot reloading stops working with an error message
```

carlosms commented on CONTRIBUTING.md:
```diff
@@ -137,6 +145,11 @@ if needed)
     you can see the progress of the build from `sourced-ui` logs (see next step).
 
+    It may happen that the hot reloading stop working with an error message
+    `Failed to compile` that can not be solved modifying the source code.
```
```suggestion
    `Failed to compile` that cannot be solved modifying the source code.
```

carlosms commented on CONTRIBUTING.md:
```diff
@@ -137,6 +145,11 @@ if needed)
 
+    It may happen that the hot reloading stop working with an error message
+    `Failed to compile` that can not be solved modifying the source code.
+    It use to happen when swiping between branches or stopping the watcher.
```
```suggestion
    It usually happens when switching between branches or stopping the watcher.
```

carlosms commented on CHANGELOG.md:
```diff
@@ -18,6 +18,10 @@
 
+### Internal
+
+- Improve docs about development mode to [run source{d} with hot reloading](./CONTRIBUTING.md#run-sourced-ce-for-development-with-hot-reloading). Previous docs didn't explain how to build `sourced-ui` development image.
```
```suggestion
- Improve docs about development mode to [run source{d} with hot reloading](./CONTRIBUTING.md#run-sourced-ce-for-development-with-hot-reloading). Previous docs didn't explain how to build the `sourced-ui` development image.
```

dpordomingo commented:
thank you for your suggestions @carlosms that were merged by 734a42c

dpordomingo commented:
I rebased and removed the collapsible effect of the `Unreleased` section as done in `sourced-ce` and the projects following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)

https://github.com/src-d/sourced-ui/pull/282/files#diff-4ac32a78649ca5bdd8e0ba38b7006a1eL4

Let me know if you'd prefer that it's kept, and I'll do so.
